### PR TITLE
CLOUDSTACK-9864 cleanup stale worker VMs after job expiry time

### PR DESCRIPTION
**Pull request by DaanHoogland**

### CODE REVIEW

File: plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java [hunk 1]
```diff
@@ -128,6 +129,7 @@
 public class VmwareManagerImpl extends ManagerBase implements VmwareManager, VmwareStorageMount, Listener, VmwareDatacenterService, Configurable {
     private static final Logger s_logger = Logger.getLogger(VmwareManagerImpl.class);
 
+    private static final long MILISECONDS_PER_MINUTE = 60000;
```
Comment (abhinandanprateek) [inline plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java]: MILI typo MILLISECONDS ..

File: plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java [hunk 2]
```diff
@@ -550,15 +552,21 @@ public boolean needRecycle(String workerTag) {
...
-         */
+        // this time-out check was disabled
+        // "until we have found out a VMware API that can check if there are pending tasks on the subject VM"
+        // but as we expire jobs and those stale worker VMs stay around untill an MS reboot we opt in to have them removed anyway
+        Long hungWorkerTimeout = 2 * (AsyncJobManagerImpl.JobExpireMinutes.value() + AsyncJobManagerImpl.JobCancelThresholdMinutes.value()) * MILISECONDS_PER_MINUTE;
+        Long letsSayNow = System.currentTimeMillis();
+        if(s_vmwareCleanOldWorderVMs.value() && letsSayNow - startTick > hungWorkerTimeout) {
+            if(s_logger.isInfoEnabled()) {
+                s_logger.info("Worker VM expired, seconds elapsed: " + (System.currentTimeMillis() - startTick) / 1000);
+            }
```
Comment (abhinandanprateek) [inline plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java]: For timeouts you may want to use java Duration, that is much cleaner.

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test centos7 vmware-55u3

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
renaming and close-opening for retests. "it works on my laptop (tm)"

**Comment by DaanHoogland** [comment]
show me the money (tm)

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
Looks like we need to rebase here as well because of the retirement of the Midonet plugin, @DaanHoogland

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test centos7 vmware-55u3

**Comment by DaanHoogland** [comment]
@karuturi @rhtyd @abhinandanprateek @swill @pdion891 can I get a witness?

**Comment by rhtyd** [comment]
LGTM

**Comment by abhinandanprateek** [comment]
LGTM on code review.

**Comment by pdion891** [comment]
I can't review much that code, but it look good and seems to fix the issue.
thanks!

**Pull request merged**